File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    answer = 0
    for i in range(0, len(nums)-2):
        for j in range(i+1, len(nums)-1):
            for k in range(j+1, len(nums)):
                checkNum = nums[i] + nums[j] + nums[k]
                if isPrime(checkNum):
                    answer += 1
    return answer

def isPrime(num):
    if num < 2:
        return False
    else:
        i = 2
        while i <= (num ** 0.5):
            if num % i == 0:
                return False
            i += 1
    return True

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        // 주어진 배열에서 서로 다른 3개의 수를 선택하기 위한 세 중첩된 반복문
        for (int i = 0; i < nums.length - 2; i++) {
            for (int j = i + 1; j < nums.length - 1; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    int sum = nums[i] + nums[j] + nums[k];
                    
                    // 소수인지 확인하는 함수 호출
                    if (isPrime(sum)) {
                        answer++;
                    }
                }
            }
        }

        return answer;
    }
    
    // 소수인지 확인하는 함수
    private boolean isPrime(int num) {
        if (num < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}
